bin/plugin/commands/performance.js

```diff
@@ -251,99 +307,57 @@ async function runPerformanceTests( branches, options ) {
-			//         ...
-			//     }
-			const confPath = `${ environmentDirectory }/.wp-env.json`;
-			const conf = { ...readJSONFile( confPath ), core: zipUrl };
-			await fs.writeFileSync(
-				confPath,
-				JSON.stringify( conf, null, 2 ),
-				'utf8'
-			);
-		}
-	}
-
-	// Printing the used folders.
-	log(
-		'\n>> Perf Tests Directory : ' +
-			formats.success( performanceTestDirectory )
-	);
-	for ( const branch of branches ) {
-		// @ts-ignore
-		const envPath = formats.success( branchDirectories[ branch ] );
-		log( `>> Environment Directory (${ branch }) : ${ envPath }` );
 	}
 
-	/*
-	 * 3- Running the tests.
-	 */
-
-	log( '\n>> Running the tests' );
+	logStep( 0, 'Looking for test files' );
 
 	const testSuites = getFilesFromDir(
-		path.join( performanceTestDirectory, 'test/performance/specs' )
-	).map( ( file ) => path.basename( file, '.spec.js' ) );
+		path.join( testRunnerDir, 'test/performance/specs' )
+	).map( ( file ) => {
+		logStep( 1, 'Found:', fAccent( file ) );
+		return path.basename( file, '.spec.js' );
+	} );
+
+	logStep( 0, 'Running tests' );
+
+	if ( wpZipUrl ) {
+		logStep( 1, 'Using:', fAccent( `WordPress v${ options.wpVersion }` ) );
+	} else {
+		logStep( 1, 'Using:', fAccent( 'WordPress trunk' ) );
+	}
 
-	const wpEnvPath = path.join(
-		performanceTestDirectory,
-		'node_modules/.bin/wp-env'
-	);
+	const wpEnvPath = path.join( testRunnerDir, 'node_modules/.bin/wp-env' );
 
 	for ( const testSuite of testSuites ) {
 		for ( let i = 1; i <= TEST_ROUNDS; i++ ) {
-			const roundInfo = `round ${ i } of ${ TEST_ROUNDS }`;
-			log( `    >> Suite: ${ testSuite } (${ roundInfo })` );
+			logStep(
+				1,
+				`Suite: ${ fAccent(
+					testSuite
+				) } (round ${ i } of ${ TEST_ROUNDS })`
+			);
 			for ( const branch of branches ) {
-				const sanitizedBranch = sanitizeBranchName( branch );
-				const runKey = `${ testSuite }_${ sanitizedBranch }_round-${ i }`;
+				logStep( 2, 'Branch:', fAccent( branch ) );
+
+				const sanitizedBranchName = sanitizeBranchName( branch );
+				const runKey = `${ testSuite }_${ sanitizedBranchName }_round-${ i }`;
 				// @ts-ignore
-				const environmentDirectory = branchDirectories[ branch ];
-				log( `        >> Branch: ${ branch }` );
-				log( '            >> Starting the environment.' );
-				await runShellScript(
-					`${ wpEnvPath } start`,
-					environmentDirectory
-				);
-				log( '            >> Running the test.' );
-				await runTestSuite(
-					testSuite,
-					performanceTestDirectory,
-					runKey
-				);
-				log( '            >> Stopping the environment' );
-				await runShellScript(
-					`${ wpEnvPath } stop`,
-					environmentDirectory
-				);
+				const envDir = branchDirs[ branch ];
+
+				logStep( 3, 'Starting environment' );
+				await runShellScript( `${ wpEnvPath } start`, envDir );
+
+				logStep( 3, 'Running tests' );
+				await runTestSuite( testSuite, testRunnerDir, runKey );
+
+				logStep( 3, 'Stopping environment' );
```

Comment:
another thought on these is that we could even pass a string and use its length for the indentation level, which would still be a little hacky, but would potentially do a more concrete job of illustrating what the first argument is doing.

```js
logStep( '>>>', 'Running tests' );
logStep( '>>>', 'Stopping environment' );
logStep( '', 'Calculating results' );
logStep( '>', 'Test suite:', … );
```

Reply:
See https://github.com/WordPress/gutenberg/pull/54309/commits/ffe445d5ecee97075c8c98535e0a3796ef1833e9